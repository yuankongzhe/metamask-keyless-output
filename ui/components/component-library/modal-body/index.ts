export { ModalBody } from './modal-body';
export type {
  ModalBodyProps,
  ModalBodyComponent,
  ModalBodyStyleUtilityProps,
} from './modal-body.types';
