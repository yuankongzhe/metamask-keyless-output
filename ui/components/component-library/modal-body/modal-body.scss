.mm-modal-body {
  max-height: 100%;
  overflow-y: auto;
  position: relative; // Allows for absolute positioning of scroll button
}
