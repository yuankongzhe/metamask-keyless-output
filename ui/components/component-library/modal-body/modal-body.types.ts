import React from 'react';

import type {
  StyleUtilityProps,
  PolymorphicComponentPropWithRef,
} from '../box';

export interface ModalBodyStyleUtilityProps extends StyleUtilityProps {
  /**
   * Additional className to add to the ModalBody
   */
  className?: string;
  /**
   * The content of the ModalBody
   */
  children?: React.ReactNode;
}

export type ModalBodyProps<C extends React.ElementType> =
  PolymorphicComponentPropWithRef<C, ModalBodyStyleUtilityProps>;

export type ModalBodyComponent = <C extends React.ElementType = 'span'>(
  props: ModalBodyProps<C>,
) => React.ReactElement | null;
