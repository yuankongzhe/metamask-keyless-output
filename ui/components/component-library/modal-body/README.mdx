import { Controls, Canvas } from '@storybook/blocks';

import * as ModalBodyStories from './modal-body.stories';

# ModalBody

`ModalBody` is a container component that handles the scrolling of the [`Modal`](?path=/docs/components-componentlibrary-modal--docs) content.

<Canvas of={ModalBodyStories.DefaultStory} />

## Props

<Controls of={ModalBodyStories.DefaultStory} />

### Children

Use the `children` prop to pass in any valid React children. The `ModalBody`'s purpose is to handle the scrolling of the `Modal` content when the content extends beyond the viewport.

<Canvas of={ModalBodyStories.Children} />

```jsx
import { ModalBody } from '../../component-library';

<div style={{ height: 100, width: 200 }}>
  <ModalBody>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vitae elit
    libero, a pharetra augue. Nullam id dolor id nibh ultricies vehicula ut id
    elit. Cras mattis consectetur purus sit amet fermentum. Donec ullamcorper
    nulla non metus auctor fringilla..
  </ModalBody>
</div>;
```

### Padding

The `ModalBody` has default padding to align with the rest of the `Modal` components. You can override this padding by passing in a `padding` prop.
Defaults:

- paddingLeft: 4 (16px)
- paddingRight: 4 (16px)
- paddingBottom: 2 (8px)
- paddingTop: 2 (8px)

<Canvas of={ModalBodyStories.Padding} />

```jsx
import {
  BackgroundColor,
  Display,
  FlexDirection,
} from '../../../helpers/constants/design-system';
import { ModalBody, Text } from '../../component-library';

<div style={{ height: 200, width: 300 }}>
  <ModalBody
    display={Display.Flex}
    flexDirection={FlexDirection.Column}
    paddingLeft={0} // removing horizontal padding from ModalBody
    paddingRight={0}
  >
    <Text paddingLeft={4} paddingRight={4}>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vitae elit
      libero, a pharetra augue. Nullam id
    </Text>
    <Text
      backgroundColor={BackgroundColor.primaryMuted}
      paddingLeft={4}
      paddingRight={4}
    >
      Element touches edge of ModalBody
    </Text>
    <Text paddingLeft={4} paddingRight={4}>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla vitae elit
      libero, a pharetra augue. Nullam id
    </Text>
  </ModalBody>
</div>;
```
