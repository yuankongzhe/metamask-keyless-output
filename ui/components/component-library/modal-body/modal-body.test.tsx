import React from 'react';
import { render } from '@testing-library/react';

import { ModalBody } from './modal-body';

describe('ModalBody', () => {
  it('should render ModalBody without error', () => {
    const { getByTestId } = render(<ModalBody data-testid="modal-body" />);
    expect(getByTestId('modal-body')).toBeDefined();
    expect(getByTestId('modal-body')).toHaveClass('mm-modal-body');
  });
  it('should match snapshot', () => {
    const { container } = render(<ModalBody />);
    expect(container).toMatchSnapshot();
  });
  it('should render with and additional className', () => {
    const { getByTestId } = render(
      <ModalBody data-testid="modal-body" className="test-class" />,
    );
    expect(getByTestId('modal-body')).toHaveClass('test-class');
  });
});
