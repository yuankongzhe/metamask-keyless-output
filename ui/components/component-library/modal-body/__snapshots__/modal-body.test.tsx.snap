// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody should match snapshot 1`] = `
<div>
  <div
    class="mm-box mm-modal-body mm-box--padding-right-4 mm-box--padding-left-4"
  />
</div>
`;
